lib/pliny/errors.rb

```diff
@@ -1,29 +1,30 @@
 module Pliny
   module Errors
     class Error < StandardError
-      attr_accessor :id
+      attr_accessor :id, :metadata
 
       def self.render(error)
         headers = { "Content-Type" => "application/json; charset=utf-8" }
-        data = { id: error.id, message: error.message }
+        data = { id: error.id, message: error.message }.merge(error.metadata)
         [error.status, headers, [MultiJson.encode(data)]]
       end
 
-      def initialize(message, id)
-        @id = id
-        super(message)
+      def initialize(options={})
+        @id = options[:id]
+        @metadata = options[:metadata] || {}
+        super(options[:message])
       end
     end
 
     class HTTPStatusError < Error
       attr_accessor :status
 
-      def initialize(message = nil, id = nil, status = nil)
-        meta    = Pliny::Errors::META[self.class]
-        message = message || meta[1] + "."
-        id      = id || meta[1].downcase.tr(' ', '_').to_sym
-        @status = status || meta[0]
-        super(message, id)
+      def initialize(options={})
```

Comment:
What do you think about changing this signature to:

``` ruby
def initialize(message, id: nil, status: nil)
  ...
```

Which the rough justification:

1. Having exceptions take a message as their first argument is a fairly widespread convention.
2. Optional arguments are safer than options hashes.